rust/examples-frontend/svc/src/index.ts

```diff
@@ -0,0 +1,338 @@
+import { Producer } from 'mediasoup-client/lib/types';
+import { ConsumerOptions } from 'mediasoup-client/lib/Consumer';
+
+type Brand<K, T> = K & { __brand: T };
+
+type ConsumerId = Brand<string, 'ConsumerId'>;
+type ProducerId = Brand<string, 'ProducerId'>;
+
+interface ServerInit {
+    action: 'Init';
+    consumerTransportOptions: TransportOptions;
+    producerTransportOptions: TransportOptions;
+    routerRtpCapabilities: RtpCapabilities;
+}
+
+interface ServerConnectedProducerTransport {
+    action: 'ConnectedProducerTransport';
+}
+
+interface ServerProduced {
+    action: 'Produced';
+    id: ProducerId;
+}
+
+interface ServerConnectedConsumerTransport {
+    action: 'ConnectedConsumerTransport';
+}
+
+interface ServerConsumed {
+    action: 'Consumed';
+    id: ConsumerId;
+    kind: MediaKind;
+    rtpParameters: RtpParameters;
+}
+
+type ServerMessage =
+    ServerInit |
+    ServerConnectedProducerTransport |
+    ServerProduced |
+    ServerConnectedConsumerTransport |
+    ServerConsumed;
+
+interface ClientInit {
+    action: 'Init';
+    rtpCapabilities: RtpCapabilities;
+}
+
+interface ClientConnectProducerTransport {
+    action: 'ConnectProducerTransport';
+    dtlsParameters: DtlsParameters;
+}
+
+interface ClientConnectConsumerTransport {
+    action: 'ConnectConsumerTransport';
+    dtlsParameters: DtlsParameters;
+}
+
+interface ClientProduce {
+    action: 'Produce';
+    kind: MediaKind;
+    rtpParameters: RtpParameters;
+}
+
+interface ClientConsume {
+    action: 'Consume';
+    producerId: ProducerId;
+}
+
+interface ClientConsumerResume {
+    action: 'ConsumerResume';
+    id: ConsumerId;
+}
+
+type ClientMessage =
+    ClientInit |
+    ClientConnectProducerTransport |
+    ClientProduce |
+    ClientConnectConsumerTransport |
+    ClientConsume |
+    ClientConsumerResume;
+
+async function init()
+{
+    const sendPreview = document.querySelector('#preview-send') as HTMLVideoElement;
+    const receivePreview = document.querySelector('#preview-receive') as HTMLVideoElement;
+
+    sendPreview.onloadedmetadata = () => sendPreview.play();
+    receivePreview.onloadedmetadata = () => receivePreview.play();
+
+    const SL = document.querySelector('#SL') as HTMLSpanElement
+    const decSL = document.querySelector('#decSL') as HTMLButtonElement;
+    const incSL = document.querySelector('#incSL') as HTMLButtonElement;
+
+    let videoProducer: Producer | null = null;
+    let maxSpatialLayer = -1
+
+    decSL.addEventListener('click', () => {
+        if (maxSpatialLayer > -1) {
+            maxSpatialLayer -=1
+            setMaxSpatialLayer()
+        }
+    });
+    incSL.addEventListener('click', () => {
+        if (videoProducer?.rtpParameters.encodings && maxSpatialLayer < videoProducer.rtpParameters.encodings.length - 1) {
+            maxSpatialLayer += 1
+            setMaxSpatialLayer()
+        }
+    });
+
+
+    const setMaxSpatialLayer = (): void => {
```

Comment:
Better take an argument with spatial layer here, it is more confusing to read otherwise.